### PR TITLE
Add Navbar Component

### PR DESCRIPTION
**Pull request by androidmini9x**
Relates #2
I Built `Navbar` Component, and I also installed tailwind library.
I didn't create `routes.js` file yet, as the names of the pages that we will need is not clear, I will add them in the next stages.

### CODE REVIEW

File: src/components/Navbar.jsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+function Navbar() {
+
+    // This is temporary state as we will work next with redux
+    const [lang, setLang] = useState("en");
+    const [search, setSearch] = useState("");
+    const isLogin = false;
+
+    const handleSearch = (e) => {
+        e.preventDefault();
+        console.log("Search...: ", search);
```
Comment (maqalaqil) [inline src/components/Navbar.jsx]: remove console log

File: src/index.js [hunk 2]
```diff
@@ -19,3 +19,10 @@ ReactDOM.render(
 // unregister() to register() below. Note this comes with some pitfalls.
 // Learn more about service workers: https://bit.ly/CRA-PWA
 serviceWorker.unregister();
+
+const btn = document.querySelector("#toggle-navbar");
```
Comment (maqalaqil) [inline src/index.js]: use single quotation

**Review by maqalaqil — CHANGES_REQUESTED** [review]
good work @androidmini9x just try fix those points 

- fix all prettier problems 
- add unit testing for navbar 
- check that navbar has the same content as figma 
- check responsiveness from 640 > 700

next time remember to add labels & reviewer & project & milestone

File: src/components/Navbar.jsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                                        <li><Link className="bg-yellow-500 hover:bg-yellow-600 py-2 px-4 block whitespace-no-wrap" to={`/request`}>School books</Link></li>
+                                        <li><Link className="bg-yellow-500 hover:bg-yellow-600 py-2 px-4 block whitespace-no-wrap" to={`/request`}>Novels</Link></li>
+                                        <li><Link className="bg-yellow-500 hover:bg-yellow-600 py-2 px-4 block whitespace-no-wrap" to={`/request`}>Exam Books</Link></li>
+                                    </ul>
+                                </div>
+                            </>}
+
+                            <Link to="/how-it-works" className="text-white hover:bg-amber-500 hover:text-white px-3 py-2 rounded-md text-sm font-medium">How it works?</Link>
+                            <Link to="/aboutus" className="text-white hover:bg-amber-500 hover:text-white px-3 py-2 rounded-md text-sm font-medium">About Us</Link>
+                            {!isLogin && <Link to="/signup" className="text-white hover:bg-amber-500 hover:text-white px-3 py-2 rounded-md text-sm font-medium">Sign up</Link>}
```
Comment (Iseluin) [inline src/components/Navbar.jsx:85]: Good work with conditional rendering. Please be ready to link this isLogin flag with actual authentication.

Comment (Iseluin) [inline src/components/Navbar.jsx]: Good work with conditional rendering. Please be ready to link this isLogin flag with actual authentication.

**Review by Iseluin — CHANGES_REQUESTED** [review]

**Review by khadijahawa — APPROVED** [review]

File: src/components/Navbar.jsx (hunk 1 above)
Comment (androidmini9x) [reply src/components/Navbar.jsx]: Done:
- ~fix all prettier problems~
- ~add unit testing for navbar~
- check that navbar has the same content as figma:
The code is almost like our figma prototype, but just wait for the teammate who chose the logo to design it and then I'll add it to the navbar
- ~check responsiveness from 640 > 700~

File: src/index.js (hunk 2 above)
Comment (androidmini9x) [reply src/index.js]: I replace them to single quotation

File: src/components/Navbar.jsx [hunk 4]
```diff
@@ -1,134 +1,394 @@
...
+                          >
+                            Books
+                          </Link>
+                        </li>
+                        <li>
+                          <Link
+                            className="bg-yellow-500 hover:bg-yellow-600 py-2 px-4 block whitespace-no-wrap"
+                            to="/donation"
+                          >
+                            Stationery
```
Comment (Iseluin) [inline src/components/Navbar.jsx]: Change Stationery to "Stationary"

File: src/components/Navbar.jsx [hunk 5]
```diff
@@ -1,134 +1,394 @@
...
+                          >
+                            Furniture
+                          </Link>
+                        </li>
+                        <li>
+                          <Link
+                            className="bg-yellow-500 hover:bg-yellow-600 py-2 px-4 block whitespace-no-wrap"
+                            to="/donation"
+                          >
+                            School books
```
Comment (Iseluin) [inline src/components/Navbar.jsx]: Please be careful with the capital letters. It should be Books.

**Review by Iseluin — CHANGES_REQUESTED** [review]

File: src/components/Navbar.jsx [hunk 6]
```diff
@@ -1,134 +1,394 @@
...
+                          >
+                            Furniture
+                          </Link>
+                        </li>
+                        <li>
+                          <Link
+                            className="bg-yellow-500 hover:bg-yellow-600 py-2 px-4 block whitespace-no-wrap"
+                            to="/request"
+                          >
+                            School books
```
Comment (Iseluin) [inline src/components/Navbar.jsx]: Same here.

**Review by Iseluin — CHANGES_REQUESTED** [review]
Navbar looks really good but there are still a few points.

- There are still some eslint warnings, please fix them. There are errors on the files that we are not going to use such as counterAPI.js and store.js. You can directly delete those files too. 
- You missed this part of the issue : 

> All routes should be a constant variable stored in the src/routes.js file and exported from there for ease of use.

I know that the page names are not decided yet but you can create a routes.js file in the src and create them anyways. We can change their names when we need to.

- Lastly, I still cannot see any tests. Please add them.

File: src/components/Navbar.jsx (hunk 4 above)
Comment (androidmini9x) [reply src/components/Navbar.jsx]: After looking up, I found that the word is correct: (Stationery) => Kırtasiye

**Comment by androidmini9x** [comment]
- I added tests.
- Re-style the code to fit instructor advise.

File: src/App.jsx [hunk 7]
```diff
@@ -1,55 +1,14 @@
-import logo from './logo.svg';
-import { Counter } from './features/counter/Counter';
 import './App.css';
+import { BrowserRouter } from 'react-router-dom';
+
+import Navbar from './components/Navbar';
```
Comment (Iseluin) [inline src/App.jsx]: Global imports should be on top of the file, if there are any local imports, you must leave a blank line after your global imports and add them after.

File: src/__tests__/Navbar.test.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    expect(window.location.pathname).toEqual('/about-us');
+
+    userEvent.click(screen.queryAllByText(/how it works?/i)[0]);
+    expect(window.location.pathname).toEqual('/how-it-works');
+
+
+
+    userEvent.click(screen.queryAllByText(/logout/i)[0]);
+    expect(window.location.pathname).toEqual('/logout');
+});
```
Comment (Iseluin) [inline src/__tests__/Navbar.test.js]: This should be in the same place where the tested component lives. So you should take the navbar component in src/components/navbar/<Your Filename> and add the test in there too.

File: src/index.js [hunk 9]
```diff
@@ -19,3 +19,10 @@ ReactDOM.render(
 // unregister() to register() below. Note this comes with some pitfalls.
 // Learn more about service workers: https://bit.ly/CRA-PWA
 serviceWorker.unregister();
+
+const btn = document.querySelector('#toggle-navbar');
+const menu = document.querySelector('#mobile-menu');
+
+btn.addEventListener('click', () => {
+  menu.classList.toggle('hidden');
+});
```
Comment (Iseluin) [inline src/index.js]: What's the purpose of those lines here ?

File: src/routes.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+  ABOUTUS: '/about-us',
+  HOWITWORKS: '/how-it-works',
+  SIGNUP: '/signup',
+  LOGIN: '/login',
+  LOGOUT: '/logout',
+  DONATION: '/donation',
+  REQUEST: '/request',
+};
+
+export default routes;
```
Comment (Iseluin) [inline src/routes.js]: Lets export them one by one, so we can just import one thing and not waste any memory. Also, for the file names with more than one word, use "_" between the words, ex: ABOUT_US, HOW_IT_WORKS etc.

**Review by Iseluin — CHANGES_REQUESTED** [review]
Good work @androidmini9x , other than the comments I left below, can you make the cursor a pointer when we click on the language options? And I still can see eslint errors and warnings when I run npm start, and it's because of the CounterAPI and the slices. We won't be needing them so you can delete them directly. 
Thank you.

File: src/index.js (hunk 9 above)
Comment (androidmini9x) [reply src/index.js]: This icon was responsible for the hamburger button, however I change this code to another code using `useRef` hook.

**Review by Iseluin — APPROVED** [review]
Amazing work @androidmini9x , really good job. I especially like your tests. There is one last little issue, the search bar looks a bit off and it closes the other tabs when opened. Can we find a workaround for that. For now, since your code is good to merge, I will merge it. But please create an issue and we can think about it later.

**Pull request merged**